testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/WaitUtils.java

```diff
@@ -162,28 +165,28 @@ public static void waitForModalFadeOut() {
         waitUntilElementIsNotPresent(By.className("modal-backdrop"));
     }
 
-    public static long getNumExecutors(KeycloakTestingClient testingClient) {
-        String numExecutors = testingClient.server().fetchString(session -> {
-            ExecutorsProvider provider = session.getProvider(ExecutorsProvider.class);
-            ExecutorService executor = provider.getExecutor("bruteforce");
-            ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) executor;
-            return threadPoolExecutor.getCompletedTaskCount();
-        });
-        return Long.valueOf(numExecutors);
-    }
-
-    public static void waitForExecutors(KeycloakTestingClient testingClient, long numExecutors) {
+    public static void waitForBruteForceExecutors(KeycloakTestingClient testingClient) {
         testingClient.server().run(session -> {
             ExecutorsProvider provider = session.getProvider(ExecutorsProvider.class);
             ExecutorService executor = provider.getExecutor("bruteforce");
             ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) executor;
-            do {
-                try {
-                    Thread.sleep(1000);
-                } catch (Exception e) {
-                }
-            } while (!threadPoolExecutor.getQueue().isEmpty());
-            assertEquals(numExecutors, threadPoolExecutor.getCompletedTaskCount());
+            try {
+                CompletableFuture.runAsync(() -> {
+                    do {
+                        try {
+                            Thread.sleep(1000);
+                        } catch (InterruptedException e) {
+                            throw new RuntimeException(e);
+                        }
+                    } while (threadPoolExecutor.getActiveCount() > 0);
```

Comment:
I think we should check !threadPoolExecutor.getQueue().isEmpty() AND threadPoolExecutor.getActiveCount() > 0 here. IIRC, executors that are in the queue are not active. So this loop could theoretically terminate while executors are still queued.